File: Examples/Reflection/ReflectionIgnoreAttribute/TypeToIgnore.cs
using KY.Generator;

namespace ReflectionIgnoreAttribute;

/// <summary>
/// [GenerateIgnore] on a type stops the type itself from being written - which is the point: the
/// TypeScript side of such a type is written by hand once and kept, instead of being overwritten on
/// every run. A member may not just use it, because there would be nothing to import; it is bound to
/// the hand written Output/type-to-ignore.ts with [GenerateImport] on <see cref="TypeToRead"/>.
/// </summary>
[GenerateIgnore]
public class TypeToIgnore
{
    public string StringProperty { get; set; } = "";
}

File: Examples/Reflection/ReflectionIgnoreAttribute/Output/type-to-ignore.ts
// Hand written on purpose - TypeToIgnore carries [GenerateIgnore], so the generator never writes this
// file and never overwrites it. [GenerateImport] on TypeToRead points the import of its members here,
// which is what keeps a member of an ignored type working: its TypeScript side stays under your control.

export class TypeToIgnore {
    public stringProperty: string = "";
    public onlyOnTheTypeScriptSide: boolean = true;
}
